### ISSUE TITLE
useIMask broken `ref` types

### ISSUE THREAD

**Issue by n1k1c4** [uNmAnNeR/imaskjs#646]
Even if you just use the example hook `useIMask` from the readme, where html `input` is used you got ts error at `ref`.
There is a ts error at `ref`:
```
Type 'MutableRefObject<ReactElement>' is not assignable to type 'LegacyRef<HTMLInputElement> | undefined'.
```

In my local project a have additional another error but can't reproduce in codesandbox:
![](https://habrastorage.org/webt/dt/ub/5f/dtub5fsnkepvz-vwkxbg6b7vqoq.png)

**To Reproduce**
https://codesandbox.io/s/react-imask-reproduce-bug-broken-ref-types-qib5k2

**Environment:**
 - OS: macOS Monterey 12.3
 - Browser: Chrome 99.0.4844.83 (Official Build) (arm64)
 - react-imask 6.4
 - react-scripts 5.0
 - typescript 4.6
 - react 17.0
 - react-dom 17.0

**Comment by andretv** [comment]
@n1k1c4 it's not a fix, but to bypass typescript type check just cast `ref` type as `LegacyRef<HTMLInputElement>`.

Like this:
```tsx
// IMask hook
const { ref, value } = useIMask(config);

// ...

return (
  <input  type='text' ref={ref as LegacyRef<HTMLInputElement>} value={value} />
)
```

I was having the same issue and came up with this to supress the error.
I hope it helps :)

**Comment by uNmAnNeR** [comment]
`HTMLInputElement` is now default but in a general case it's better to use `useIMask<IMask.AnyMaskedOptions, HTMLInputElement>`

**Comment by pawlarius** [comment]
Hi @uNmAnNeR , how to use `useIMask<IMask.AnyMaskedOptions, HTMLInputElement>`?
The typescript will fail here.

<img width="985" alt="Screen Shot 2022-09-19 at 12 05 45" src="https://user-images.githubusercontent.com/27823957/190952811-0f4c4423-facf-4797-8d8a-fb08820d1c6a.png">


I'm using react-imask 6.4.2, but the default ref type is not `HTMLInputElement`, but it's using custom `ReactElement` from `react-imask/dist/mixin`.

<img width="395" alt="Screen Shot 2022-09-19 at 12 08 24" src="https://user-images.githubusercontent.com/27823957/190953006-8a65fa57-1671-4e44-853f-e3b2a7f32bba.png">

**Comment by uNmAnNeR** [comment]
@pawlarius it's fixed in v6.4.3 which is not released yet
